### DISCUSSION TITLE
Video analysis with sktime

### DISCUSSION THREAD

**Discussion by qualityland27** [alan-turing-institute/sktime#1386] [Q&A]
Hi together, 

I am working on a project using mediapipe to extract body landmarks of a person within a picture. Based on that data  (maybe some additional data about objects in the picture) I want to do classification of actions within a video. Is sktime good choice for  a task like that?

Thanks!

**Comment by fkiraly** [comment]
`sktime` does not support image/video learning tasks, but you can combine image/video feature extraction of time series features (e.g., position of landmark over time) and then use time series classification for action classification. In fact, motion or action classification is a "typical" time series classification task, after such a step.

**Reply by fkiraly** [reply]
@TonyBagnall might also have some pointers here.

**Reply by qualityland27** [reply]
Thanks for your Answer @fkiraly ! By "position of landmark over time" you mean sth like that?

![Bildschirmfoto vom 2021-09-06 13-07-08](https://user-images.githubusercontent.com/51876194/132208248-2dd1b1d9-96dc-44aa-89d5-533d719bb6b8.png)

Within one row there is the class in the first column followed by  x and y cooridnates of several landmarks. Each Row is one datapoint within the series.

**Reply by fkiraly** [reply]
Yes, precisely!

Though note: you'll need the "class" column to be separate, and the panel (the collection of all series) needs to be in one of the `sktime` compatible data representations. Check the notebook `AA_datatypes_and_datasets` for the format specifications here (panel data section), and the classification notebooks for current example workflows.

You'll also want to get familiar with the converter functionality (same `AA_datatypes` notebook) because time series classifiers are not fully refactored yet to cope with the compatible data type representations. You'll want the `numpy3D` or the `nested_univ` format before using the classifiers which are not fully refactored. See here https://github.com/alan-turing-institute/sktime/issues/1146 for the status of the refactor.

If you have data frames like the above: the easiest way is perhaps to put them in a list (without the `class column`, then use `convert_to` to get to `numpy3D` or `nested_univ` if you need to.

**Reply by fkiraly** [reply]
The notebooks are either [on binder](https://mybinder.org/v2/gh/alan-turing-institute/sktime/main?filepath=examples) for playing around in your browser, or in the [examples folder](https://github.com/alan-turing-institute/sktime/tree/main/examples) for reading or running locally.

**Reply by qualityland27** [reply]
Thanks again for your answer @fkiraly ! 

With your help I was able to train a sktime classifier. In my jupyter notebook I am able to make predictions. Works fine so far.

Now I am trying to implement a python script for using the classifier in my system. You wisely pointed out to ` AA_datatypes`. But I still have an issue with the datatype..

**Training in jupyter notebook:**
I summarized all columns (except "class") into a new column named "dim_0". I did this using ``df['dim_0'] = df.apply(lambda r: tuple(r), axis=1).apply(np.array)``. Then I selected this column as X (features) to train classifier. See picture:
![image](https://user-images.githubusercontent.com/51876194/135511831-2f243d61-d49e-4dcc-a751-ad5f5dd8e71a.png)

_note: Prediction with `y_pred = tsf.predict(X_test)` works fine!_

**Classification in pycharm with live data:**
I did exact the same I did for training. I putted all columns from the dataframe together and create a new column named 'dim_0' using:
`df['dim_0'] = df.apply(lambda r: tuple(r), axis=1).apply(np.array)`

- select new column with:
`df = df[['dim_0']]`

- do prediction with:
`predicted_class = sktimeModel.predict(df)`

Error Message:
ValueError: Input contains NaN, infinity or a value too large for dtype('float32').


Afterwards, I tried following your recommendation: I putted the features into a list and used `convert_to` to get `numpy3D`.

Error Message:
TypeError: No valid mtype could be identified


I tried very long to understand the problem but I was not able to. So I want to ask again for your help. Thanks!
